integration/helpers.go

```diff
@@ -54,6 +54,23 @@ func parseOutput[T any](output string, normalize func(*T)) ([]*T, error) {
 	return ret, nil
 }
 
+func UnpackJsonArray(output string) (string, error) {
+	var objects []map[string]any
```

Comment:
Just for understanding why do we need a map here? I see we are ignoring the keys when merging the objects?

Reply:
This is done in order to handle any type of input in the form of `[{...},{...}]`. Just an abstract way of telling that we have an array of JSON objects and we done care what they are.
We are going to extract these objects and write them in their own line

The validation will happen of the type will happen later in `parseOutput`.

This is inefficient since we are unmarshalling the JSON twice (and marshalling it an additional time)
That is why I asked if it would be better when we have two functions `ExpectEntriesToMatch` which handle the JSON output differently or how else we could generalize it. Like trying both methods

Reply:
Yeah, I propose to have two functions. 

The existing one, for which we should be more specific in the description. Maybe something like this:

```golang
// ExpectEntriesToMatch verifies that all the entries in expectedEntries are
// matched by at least one entry in the output (Independent JSON objects).
func ExpectEntriesToMatch[T any](output string, normalize func(*T), expectedEntries ...*T) error
```

And a new one:
```golang
// ExpectEntriesInArrayToMatch verifies that all the entries in expectedEntries
// are matched by at least one entry in the output (JSON array).
func ExpectEntriesInArrayToMatch[T any](output string, normalize func(*T), expectedEntries ...*T) error
```

WDYT?

Reply:
I was thinking if we can use something like [this](https://stackoverflow.com/a/55014220/7274380) but that might be bit magic? :smile:  Also, sounds good having two functions, makes it more clear!

Reply:
That was also my thought :D Magic or not.
Some people like it - others not. But i also think its more beneficial to have it clearer

Reply:
Mmmm, IMO, it's non that "magic" 😝. However, I consider that having two helpers would make more evident (during the development and reveiwing) a change to the output format of any of the gadgets (array to indentent objects, or viceversa). In genereal, I think it is a good practice to not have generic things on the testing infrastructure, so that one easily realized about (desired or indesired) changes introduced with a PR. I don't know, I have always have this idea, but I maybe it is just me and my "control mania" 😆